util/webhook/webhook.go

```diff
@@ -342,11 +351,13 @@ func (a *ArgoCDWebhookHandler) Handler(w http.ResponseWriter, r *http.Request) {
 		payload, err = a.bitbucketserver.Parse(r, bitbucketserver.RepositoryReferenceChangedEvent)
 	default:
 		log.Debug("Ignoring unknown webhook event")
+		replyBadRequest(w, "Unknown webhook event")
```

Comment:
I think it would be good to use the more standardized method `http.Error()` to write the status to the client.